pkg/checks/checks.go

```diff
@@ -32,6 +32,9 @@ func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urlto
 		resp, err := cli.Do(req)
 		if err, ok := err.(*url.Error); ok {
 			if err, ok := err.Err.(*net.OpError); ok {
+				if _, ok := err.Err.(*net.DNSError); ok {
```

Comment:
I dont like it as it might be valid error, but it fails because now with this it start fast and this record does not exist at that time... when moved to pod we might want to remove it and let pod to handle this?

Reply:
In the ARM template, I think you should pass in the first part of the FQDN into the apiserver public IP so that it is actually registered.  Then maybe this error catch isn't necessary.
You also need to check with @amanohar whether OSA will have its own separate dns domain which the RP will select a pre-created name from and how to handle that.